src/main/java/scripts/uploadscripts/SummariesToBigQuery.java

```diff
@@ -0,0 +1,231 @@
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.concurrent.ConcurrentHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SuppressFBWarnings(
+    value = {"NP_UNWRITTEN_PUBLIC_OR_PROTECTED_FIELD"},
+    justification =
+        "The statistics field will be populated by sub-classes of MeasurementCollectionScript.")
+public class SummariesToBigQuery extends UploadScript {
+  private static final Logger logger = LoggerFactory.getLogger(SummariesToBigQuery.class);
+
+  /** Public constructor so that this class can be instantiated via reflection. */
+  public SummariesToBigQuery() {}
+
+  protected String projectId; // google project id
+  protected String datasetName; // big query dataset name
+
+  protected TestConfiguration renderedTestConfiguration;
+  protected TestRunner.TestRunSummary testRunSummary;
+  protected MeasurementCollectionScript.MeasurementResultSummary[] measurementCollectionSummaries;
+
+  /**
+   * Setter for any parameters required by the upload script. These parameters will be set by the
+   * Result Uploader based on the current Upload List, and can be used by the upload script methods.
+   *
+   * @param parameters list of string parameters supplied by the upload list
+   */
+  @Override
+  public void setParameters(List<String> parameters) throws Exception {
+    if (parameters == null || parameters.size() < 2) {
+      throw new IllegalArgumentException(
+          "Must provide BigQuery project_id and dataset_name in the parameters list");
+    }
+    projectId = parameters.get(0);
+    datasetName = parameters.get(1);
+  }
+
+  private static String testRunTableName = "testRun";
+  private static String testScriptResultsTableName = "testScriptResults";
+  private static String measurementCollectionTableName = "measurementCollection";
+
+  /**
+   * Upload the test results saved to the given directory. Results may include Test Runner
+   * client-side output and any relevant measurements collected.
+   */
+  @Override
+  public void uploadResults(
+      Path outputDirectory, ServiceAccountSpecification uploaderServiceAccount) throws Exception {
+    // get a BigQuery client object
+    logger.debug("BigQuery project_id:dataset_name: {}:{}", projectId, datasetName);
+    BigQuery bigQueryClient =
+        BigQueryUtils.getClientForServiceAccount(uploaderServiceAccount, projectId);
+    // read in TestConfiguration TestRunSummary, and array of
+    // MeasurementCollectionScript.MeasurementResultSummary objects
+    renderedTestConfiguration = TestRunner.getRenderedTestConfiguration(outputDirectory);
+    testRunSummary = TestRunner.getTestRunSummary(outputDirectory);
+    measurementCollectionSummaries =
+        MeasurementCollector.getMeasurementCollectionSummaries(outputDirectory);
+
+    // insert a single row into testRun
+    if (BigQueryUtils.checkRowExists(
+        bigQueryClient, projectId, datasetName, testRunTableName, "id", testRunSummary.id)) {
+      logger.warn(
```

Comment:
Why check for this condition only to keep going?

Reply:
Just so we log the warning that these results have already been uploaded.